src/task/mod.rs

```diff
@@ -148,6 +148,8 @@ cfg_default! {
     mod block_on;
     mod builder;
     mod current;
+    #[cfg(not(target_os = "unknown"))]
```

Comment:
This also needs to not be enabled when we build without the `default` feature flag, maybe it would be good to introduce an explicit `executor` feature, which groups the deps and this together

Reply:
Why so? I'm not sure I follow here. All of this module requires a runtime, thus default to be enabled?

Can you point to a code path which would need it?

Reply:
`cargo build --no-default-featurs --features unstable` should work, but I think will not work with the current setup

Reply:
It already fails on master (#842) and the failures are the same with this PR, it doesn't add more failures

Reply:
I'll be more than happy to fix things if I'm wrong once master is fixed but I don't think it should block this PR

Reply:
Build fine now

Reply:
Thank you! Can you do me a favor and add a ci check for this while we are here? otherwise this will break again and again.

Reply:
Done!